### ISSUE TITLE
Initial code merge of pywbemcli

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbemtools#2]
This issue defines the initial code commit for pywbemcli

**DONE:**  I will call this prototype until we agree basic structure is OK.
Prototype with the following groups and subcommands:
    class - except for invoke
    instance - except for invoke, create, modify, delete
    qualifier - get and enumerate functions done.
    server - namespaces, interop, brand, info, profiles

This is based on the initial prototype branch in issue#652 of pywbem repository and rework.

**DISCUSSION:** Is this the correct name for the repository or should it be more general so we can add other tools, ex. graphical browser, etc.???

**Possible TODO List per Karl**. I am moving this TODO list to the issue before next telecon but it was my first cut last night when I did the first branch commit.  

**Discussion:** at what point should we commit the pr for this and create another one???

1. Add unittests for functions. Required

2.  Add mock test capability. Currently I was thinking of mock_cim_operations as the logical mock interface for tools like this. Required

**DISCUSSION:** What are logical alternatives?

3. Add unittests against server (right now I have a small script test_help.sh and test_ops.sh just to see that they are running). Required. However, I would like to discuss possibilities for testing since the tools in click appear to be NOT helpful.  Would like to be able to actually run script and look as result in mock and against server.

4. Finish existing functions in instance (invokemethod, create, modify) Required

5. Documentation and help cleanup. Required

6.  Consider some others instance operations that have proven viable for script testing environments (match( match an instance created in client against one in server. We called this test in cimcli), equals, etc.) Optional

6a. Add at least table and txt display options code.

7. finish invokemethod to class. Need to create the param parser for that. Required

8. Add subscriptions command group (list, create, delete). I have not yet figured out how this ties to an indication consumer but probably we should be able to attach one in at least some modes of operation. Required

9. Add profiles command group - Not sure exactly what will be in this yet but at least start with overview of a profile. Optional

8. Gathering time on operations - We need to answer several questions to gather time on cmd execution. There are a number of alternatives for this.  See design document, Optional. 

**DISCUSSION:** Alternative means to gather time information.

9. So far we have gotten away without having a config file.  Weeed to review the need.  However, I think we will hit that wall soon so that there will be things that a) a user wants semipermanent, a group wants permanent. Optional

10. Reorganize code more logically (should we separate the click functions from the cmd executors)?
**DISCUSSION:** Would appreciate Andy's comments on organization of the code. Required.

11. Sort out how to use the spinner.  Right now it interferes with output. **DISCUSSION**

12. Format for some of the boolean options.  Most booleans are negative unless set (i.e. -v  for verbose). However, include qualifiers is messy in that  for classes the default is false but the normal usage is true and for instances, the normal usage is false.  
```
  -l, --localonly                 Show only local properties of the class.
  --includequalifiers / --no_includequalifiers
                                  Include qualifiers in the result. Default is
                                  to include qualifiers
```
I don't like this
**TODOs beyond initial commit**
1. Add functionality for a base of servers/connections so that the user only needs to know a name and all of the details of the connection are in the base:

A user would define and name a connection and from then on could work with that connection simply using the name instead of the details of the connection.  This means keeping the defined details in a base somewhere (database, txt or csv file, or the config file).
Example:
* connections
   * list - list connections in base
   * select - select a predefined connection by name, pick from list, etc. This becomes the current connection.
   * create - user provides parameters (url, default ns, user, pw, etc. and a name) and a new connection is created and saved

A new option on the cmd line would define the connection for pywbemcli in terms of the name of an existing definition, not all of the details.

2. Extend instance create/modify instances to accommodate embedded instances.

**Comment by KSchopmeyer** [comment]
Notes from Friday

1. Lets rename this to pywbemtools.

2. Karl asked for overall code structure review before we vote on merge of this branch.

3. Agreed that we want basic cmds running with tests to as basis for for commit of this branch. Andy noted that we want to avoid massive commits.

4. We discussed how to add iteration time to pywbem like but agreed that this should be part of pywbem itself, not pywbem like.

5. Andy suggested that we need to look more into function mocking, probably at the level of CIM o0erations and possibly in pywbem to provide mocking tool for pywbem like

6. Karl demoed running cmds and we discussed possible issues with spinner. No nresolution.

7. Discussed wired forms of the include qualifier option but no resolution

8. Agreed that documentation for pywbem like should be separate from pywbem with just cross links
